_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,177 @@
+---
+name: IFrame has a unique accessible name
+
+description: |
+ Each iframe element has a unique accessible name
+
+success_criterion:
+- 4.1.2
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to `iframe` elements within a given [document context](#document-context) and [exposed to assistive technologies](#exposed-to-assistive-technologies) that have a [non-empty](#non-empty) accesible name.
```

Comment:
What happens with iframes within loaded iframe content? If I understand correctly, the current definition of "document context" implicitly allows a parent iframe to be the document context. That means that frame names must be unique within these branches but not necessarily across the entire tree of the root document. If my understanding is incorrect then the "document context" definition may need to be more precise about this. If my understanding is correct, then we should add a note here about applicability to iframes within loaded iframe content. We should probably have corresponding test cases too.

Reply:
@nitedog your understanding is correct, the rule applies to `iframes` with in a document context, and not for the entire tree of the root document.

I shall add an applicability and relevant test cases.